src/wallet/rpcwallet.cpp

```diff
@@ -440,14 +442,113 @@ static void SendMoney(CWallet * const pwallet, const CTxDestination &address, CA
     }
 }
 
+UniValue yak(const JSONRPCRequest& request)
+{
+    CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+            "yak \"msg\" amount\n"
+            "\nPlace any data on the blockchain.\n"
+            + HelpRequiringPassphrase(pwallet) +
+            "\nArguments:\n"
+            "1. \"yak\"                (string, optional) Add an optional message to the blockchain\n"
+            "\nResult:\n"
+            "\"txid\"                  (string) The transaction id.\n"
+            "\nExamples:\n"
+            + HelpExampleCli("yak", "\"LEr4HnaefWYHbMGXcFp2Po1NPRUeIk8km2\"")
+        );
+
+    std::string dataMsg = request.params[0].get_str();
+
+    static const char* const lut = "0123456789ABCDEF";
+    size_t len = dataMsg.length();
+
+    std::string output;
+    output.reserve(2 * len);
+    for (size_t i = 0; i < len; ++i)
+    {
+        const unsigned char c = dataMsg[i];
+        output.push_back(lut[c >> 4]);
+        output.push_back(lut[c & 15]);
+    }
+    dataMsg = output;
+
+    if((dataMsg.length() / 2) >= (MAX_OP_RETURN_RELAY - 2)){
```

Comment:
80 bytes ATM, we could increase this, but that will cause some client to not broadcast the transaction?